### DISCUSSION TITLE
Is Upptime against the GitHub TOS?

### DISCUSSION THREAD

**Discussion by dillfrescott** [upptime/upptime#749] [Q&A] [answered]
Ive heard that it is against their terms of service to host upptime using github actions

**Comment by dillfrescott** [comment]
https://news.ycombinator.com/item?id=31929308

**Reply by realpixelcode** [reply]
Some random person's comment “It's also against their ToS” without further explanation is completely worthless.

If it actually was against the ToS, why has this repo not been removed yet?

**Reply by dillfrescott** [reply]
Maybe they havent seen it

**Comment by AnandChowdhary** [comment, marked as answer]
According to [GitHub Terms for Additional Products and Features #Actions](https://docs.github.com/en/site-policy/github-terms/github-terms-for-additional-products-and-features#actions), you are not allowed to use GitHub Actions for:

- cryptomining;
- disrupting, gaining, or attempting to gain unauthorized access to, any service, device, data, account, or network (other than those authorized by the [GitHub Bug Bounty program](https://bounty.github.com/));
- the provision of a stand-alone or integrated application or service offering the Actions product or service, or any elements of the Actions product or service, for commercial purposes;
- any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden); or
- if using GitHub-hosted runners, any other activity unrelated to the production, testing, deployment, or publication of the software project associated with the repository where GitHub Actions are used.

The only place it may be an issue is if we fall under point 4 (burden on our servers), but it's up to GitHub to determine that, and they haven't told us any such thing yet. If they do, I will of course make any necessary changes to prevent any issues; I <3 GitHub and GitHub Actions, and want to cause no harm.

In fact, **Upptime uses almost 450,000 minutes of free CI time every month** as one of the most popular GitHub Actions-based template repositories on GitHub, and I shared this fact in my talk at GitHub Universe 2022: https://youtu.be/F3dtaDJi_bo?t=883
